src/components/templates/AnimateMap/components/FirebarrelWidget/FirebarrelWidget.tsx

```diff
@@ -315,9 +318,25 @@ export const FirebarrelWidget: React.FC<FirebarrelWidgetProps> = ({
 
   if (!token) return null;
 
+  const onExitClick = () => {
+    if (room) {
+      room.disconnect();
+
+      if (onExit) {
+        onExit();
+      }
+    }
```

Comment:
wrap into useCallback + 
```suggestion
    if (!room) return;
    
    room.disconnect();

    if (!onExit) return;
    
    onExit();
```